app/js/user.js

```diff
@@ -88,6 +87,49 @@
           });
         });
       });
+    },
+
+    /**
+     * Updates the local user data as well as last sync time.
+     * @param {object} user_data - The ammended user data and settings i.e.
+     * user_data = {
+     *   user: user,
+     *   settings, settings
+     * };
+     * @returns The updated user object from indexedDB
+     */
+    update_user: function(user_data) {
+      SumoDB.update_user(user_data).then(function(user) {
+        user.last_sync = Date.now();
+        return asyncStorage.setItem('user', user).then(function() {
```

Comment:
I am not sure we do. I was going to pass it through however, one of the problems was that it would overwrite some of the settings sent to the server with the hard coded values. I could check if the field has a value and only substitute with the hard coded value if it does not, but thought meh, we want to keep the object structure the same and the user will, after the profile update, definitely have a username and display_name so skip it.

With that said though, we probably need some validation on the display name field else the user could change it to be nothing and save that.

What do you think, pass it through normalize user or not?

One thing I did think about is, currently for some fields such as handset_type and operator, we have no corresponding field on the server side user profile so, these values are currently thrown away so, perhaps for now, best to still run it through normalize user. I will wait to see what you think though.